### PR TITLE
Rework parameter count inspection (fixes #2115)

### LINKED ISSUE #2115: Rework count mismatch inspection
```go
package main

import (
	"fmt"
	"reflect"
)

type b func(intParam int)
type a b

func c() {
	aFunc := generateFunc()
	if f, ok := aFunc.(a); ok {
		fmt.Println(reflect.TypeOf(f))
		f(<caret>) // should be not enough arguments in call to f
	}
}

func generateFunc() interface{} {
	return a(func (intParam int) { })
}

func main() {
	c()
}
```

See com.goide.editor.GoParameterInfoHandler#showParameterInfo

### PR DESCRIPTION
**Pull request by dlsniper**
@ignatov not sure if this is what you had in mind but let me know if I need to rework anything else

### CODE REVIEW

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -225,4 +225,4 @@
     </option>
     <option name="USE_PER_PROJECT_SETTINGS" value="true" />
   </component>
-</project>
\ No newline at end of file
+</project>
```
Comment (ignatov) [inline .idea/codeStyleSettings.xml]: Hmm, it's definitely from another change list.

File: gradle.properties [hunk 2]
```diff
@@ -26,4 +26,4 @@ sources=true
 publishUsername=user
 publishPassword=password
 publishChannel=
-org.gradle.jvmargs=-Xms256m -Xmx1024m
\ No newline at end of file
+org.gradle.jvmargs=-Xms256m -Xmx1024m
```
Comment (ignatov) [inline gradle.properties]: And this one.

**Comment by dlsniper** [comment]
Removed unrelated misc file changes

**Comment by dlsniper** [comment]
Once this gets merged I plan to work on fixing: https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/1968 as currently this PR doesn't cover it. However I want to keep the PR smaller than the one with assignment count mismatch so I'd rather handle one issue at a time as it appears this time around. Thank you

**Comment by dlsniper** [comment]
Updated to be mergeable again (I'll update it as soon as it changes state if again).

**Comment by dlsniper** [comment]
Updated

**Comment by dlsniper** [comment]
Updated

**Comment by dlsniper** [comment]
Updated this as well.

File: testData/highlighting/funcCall.go [hunk 3]
```diff
@@ -143,8 +143,8 @@ func bar_m() (int, int) {
 	return 0, 0
 }
 
-func <warning>main127</warning>() {
-	foo_m<error>(bar_m())</error>
+func <warning descr="Unused function 'main127'">main127</warning>() {
```
Comment (ignatov) [inline testData/highlighting/funcCall.go]: Please don't add such markup without need, it's very hard to review.

File: tests/com/goide/inspections/GoHighlightingTest.java [hunk 4]
```diff
@@ -135,6 +135,7 @@ protected String getBasePath() {
   public void testEmbeddedInterfacePointer()  { doTest(); }
   public void testPlaceholderCount()          { doTest(); }
   public void testTypeConversion()            { doTest(); }
+  public void testGh2115()                    { doTest(); }
```
Comment (ignatov) [inline tests/com/goide/inspections/GoHighlightingTest.java]: Better to create a test with a meaningful name, not a issue number.

Comment (dlsniper) [inline tests/com/goide/inspections/GoHighlightingTest.java]: I'll update it, yes.

File: testData/highlighting/funcCall.go (hunk 3 above)
Comment (dlsniper) [inline testData/highlighting/funcCall.go]: I'll try and remove those, I've understood meanwhile why they appear in the tests since I've wrote this :)

**Comment by dlsniper** [comment]
Done, I've fixed the previous comments, added the extra test cases from the other two related issues and made the suggestion @zolotov had regarding the position of where the error is reported, it's now on the function name not the the whole arguments list (which makes it more visible when there are no arguments).

**Comment by dlsniper** [comment]
The only thing that's different is the error message vs the Go compiler in some cases but I think it's more readable this way (I can use the compiler messages if needed).

**Pull request closed without merging**